packages/block-library/src/utils/use-template-context.js

```diff
@@ -0,0 +1,310 @@
+	// Check for taxonomy patterns
+	// e.g. taxonomy-product-category, taxonomy-product-category-electronics
+	if ( templateSlug.startsWith( 'taxonomy-' ) ) {
+		const taxonomyPart = templateSlug.substring( 9 );
+		const taxonomyParts = taxonomyPart.split( '-' );
+
+		if ( taxonomyParts.length === 1 ) {
+			// No dashes, so the entire part is the taxonomy
+			// e.g. taxonomy-product -> taxonomy: "product", term: null
+			return {
+				...DEFAULT_CONTEXT,
+				taxonomy: taxonomyPart,
+			};
+		}
+
+		if ( taxonomyParts.length === 2 ) {
+			// Single dash
+			// e.g. taxonomy-product-category -> taxonomy: "product-category", term: null
+			return {
+				...DEFAULT_CONTEXT,
+				taxonomy: taxonomyPart,
+			};
+		}
+
+		// Multiple dashes
+		// e.g. taxonomy-product-category-electronics -> taxonomy: "product-category", term: "electronics"
+		const taxonomy = taxonomyParts.slice( 0, -1 ).join( '-' );
+		const termSlug = taxonomyParts[ taxonomyParts.length - 1 ];
+
+		return {
+			...DEFAULT_CONTEXT,
+			taxonomy,
+			termSlug,
+		};
+	}
+
+	// Check for built-in taxonomy patterns
+	// e.g. category, tag
+	if ( templateSlug === 'category' ) {
+		return {
+			...DEFAULT_CONTEXT,
+			taxonomy: 'category',
+		};
+	}
+	if ( templateSlug === 'tag' ) {
+		return {
+			...DEFAULT_CONTEXT,
+			taxonomy: 'post_tag',
+		};
+	}
+
+	// Check for specific term patterns for built-in taxonomies
+	// e.g. category-news, tag-featured
+	if ( templateSlug.startsWith( 'category-' ) ) {
+		return {
+			...DEFAULT_CONTEXT,
+			taxonomy: 'category',
+			termSlug: templateSlug.substring( 9 ),
+		};
+	}
+	if ( templateSlug.startsWith( 'tag-' ) ) {
+		return {
+			...DEFAULT_CONTEXT,
+			taxonomy: 'post_tag',
+			termSlug: templateSlug.substring( 4 ),
+		};
+	}
+
+	return DEFAULT_CONTEXT;
+}
+
+/**
+ * Template slug parser that validates taxonomies exist.
+ *
+ * @param {string}   templateSlug The template slug to parse.
+ * @param {Function} getTaxonomy  Function to get taxonomy record.
+ * @return {Object} Object containing taxonomy, termSlug, isAuthor, and authorSlug.
+ */
+export function parseTemplateSlugWithValidation( templateSlug, getTaxonomy ) {
+	if ( ! templateSlug ) {
+		return DEFAULT_CONTEXT;
+	}
+
+	if ( templateSlug.startsWith( 'taxonomy-' ) ) {
+		const taxonomyPart = templateSlug.substring( 9 );
+		const taxonomyParts = taxonomyPart.split( '-' );
+
+		// Check if the entire string is a valid taxonomy.
+		const fullTaxonomyRecord = getTaxonomy( taxonomyPart );
+		if ( fullTaxonomyRecord ) {
+			return {
+				...DEFAULT_CONTEXT,
+				taxonomy: taxonomyPart,
+			};
+		}
+
+		// Try splitting the taxonomy into parts to find a valid taxonomy.
+		for ( let i = taxonomyParts.length - 1; i >= 1; i-- ) {
+			const potentialTaxonomy = taxonomyParts.slice( 0, i ).join( '-' );
+			const potentialTermSlug = taxonomyParts.slice( i ).join( '-' );
+
+			const taxonomyRecord = getTaxonomy( potentialTaxonomy );
+			if ( taxonomyRecord ) {
+				return {
+					...DEFAULT_CONTEXT,
```

Comment:
I don't think we need to spread `DEFAULT_CONTEXT` at all when returning values, do we? We could even delete it and return an empty object when we return just that, since all the values are falsy.

Reply:
Deleting it works well, thanks! Done in https://github.com/WordPress/gutenberg/pull/72549/commits/6d7e4478cee291d48e90dee6b556f2b2fa05504e.